packages/api-explorer/src/components/SideNav/SideNav.tsx

```diff
@@ -113,19 +113,20 @@ export const SideNav: FC<SideNavProps> = ({ headless = false, spec }) => {
     const searchParams = new URLSearchParams(location.search)
     if (debouncedPattern && debouncedPattern !== searchParams.get('s')) {
       searchParams.set('s', debouncedPattern)
-      navigate(location.pathname, { search: searchParams.toString() })
+      navigate(location.pathname, { s: searchParams.get('s') })
     } else if (!debouncedPattern && searchParams.get('s')) {
       searchParams.delete('s')
-      navigate(location.pathname, { search: searchParams.toString() })
+      navigate(location.pathname, { s: null })
     }
-  }, [location.search, debouncedPattern])
+  }, [debouncedPattern])
 
   useEffect(() => {
     let results
     let newTags
     let newTypes
     let newTypeTags
     const api = spec.api || ({} as ApiModel)
+    setSearchPattern(searchPattern)
```

Comment:
I have to remove `location.search` as a dependency from the useEffect and  setSearchPattern(searchPattern) in the searchPattern listener for routing backwards/forwards. Imagine there you have it blank, then search ‘test’ in the box. If you press the back button, APIExplorer will have its listener trigger and set our search pattern to ‘’.  *The problem, however, if `location.search` is a dependency of the useEffect based on `debouncedPattern`, the pattern will not have debounced immediately and the `debouncedPattern` will still point to ‘test’. Hence the error.

Why we need `setSearchPattern(searchPattern`: once APIExplorer executes `setSearchPatternAction`, this useEffect will only update the filters that will be visible in the sideNav. We need the search bar to also reflect the new search change, if that makes sense.